features/events/daemon/src/main/java/org/opennms/netmgt/eventd/DefaultEventHandlerImpl.java

```diff
@@ -119,6 +123,17 @@ public void run() {
             }
 
             for (final Event event : events.getEventCollection()) {
+                if (event.getNodeid() == 0) {
+                    final Parm foreignSource = event.getParm("_foreignSource");
+                    final Parm foreignId = event.getParm("_foreignId");
```

Comment:
If `foreignSource` was not found, or has a null value, we don't need to continue search for the the `foreignId`.